pkg/destroy/aws/aws.go

```diff
@@ -1269,6 +1271,24 @@ func deleteEC2VPCEndpointsByVPC(client *ec2.EC2, vpc string, failFast bool, logg
 	return nil
 }
 
+func deleteEC2VPCPeeringConnection(client *ec2.EC2, id string, logger logrus.FieldLogger) error {
+	_, err := client.DeleteVpcPeeringConnection(&ec2.DeleteVpcPeeringConnectionInput{
+		VpcPeeringConnectionId: &id,
+	})
+	if err != nil {
+		switch err.(awserr.Error).Code() {
+		case "InvalidVpcPeeringConnectionID.NotFound":
+			return nil
+		case "UnauthorizedOperation":
```

Comment:
This doesn't seem like the right behavior, if the installer is supposed to deleted the peering object, it can't skip on unauthorized.

Reply:
What would you propose as the correct behavior? These are the other options that I can think of.
1. The destroyer always ignores the VPC Peering Connection. This may lead to leakage. The VPC Peering Connection will be created in-cluster, so there is no external entity to take responsibility for deleting the VPC Peering Connection.
2. The destroyer fails indefinitely if the user does not have the necessary permissions. The user may not be able to resolve this if they cannot get the permissions. Note that permissions to create a VPC Peering Connection does not imply permissions to delete a VPC Peering Connection.
3. Increase the permissions required by the installer to include permissions to delete a VPC Peering Connection.

Reply:
> What would you propose as the correct behavior? These are the other options that I can think of.
> 
>     1. The destroyer always ignores the VPC Peering Connection. This may lead to leakage. The VPC Peering Connection will be created in-cluster, so there is no external entity to take responsibility for deleting the VPC Peering Connection.

The goal for installer destroy is to delete any resource created as an effect of openshift core operators. this resource is NOT created by one, rather is an user-installed resource. I don't think it's the responsibility for the installer to clean this up.

the installer CAN clean the resource as long as it's tagged correctly, but that's only case by case and it's going to be best effort.. i.e.

- the user needs to take the responsibility of opting into cleanup by tagging
- the user needs to take responsibility of making sure the installer CAN actually delete this due  to restrictions/permissions.

> 
>     2. The destroyer fails indefinitely if the user does not have the necessary permissions. The user may not be able to resolve this if they cannot get the permissions. Note that permissions to create a VPC Peering Connection does not imply permissions to delete a VPC Peering Connection. For the Hive use case, there would need to be special code which parses the destroyer logs to determine that the only thing blocking the uninstall from completing is a stuck VPC Peering Connection.

personally, hive as any other user needs to manage day-2 resources. the installer can most definitely adds this capability to take this off, but there shouldn't be expectation of safety. the user here, hive, needs to do the diligence of making sure that installer will have enough permissions.

> 
>     3. Increase the permissions required by the installer to include permissions to delete a VPC Peering Connection.

That wouldn't be correct because the vpc peer connection deletion is not IMPORTANT to openshift install variants.

Reply:
> the installer CAN clean the resource as long as it's tagged correctly, but that's only case by case and it's going to be best effort.. i.e.
>
>    the user needs to take the responsibility of opting into cleanup by tagging
>    the user needs to take responsibility of making sure the installer CAN actually delete this due to restrictions/permissions.

In my view, this exactly what this PR is doing. The installer CAN clean the resource if it is tagged correctly. It attempts to. If it cannot, then it ignores it. The installer is not taking responsibility for destroying the resource: The installer is making a best-effort attempt.

Reply:
That being said, if you feel that it would be better for the installer to not make ANY attempt to destroy the VPC Peering Connections, then I can live with that, too.

Reply:
if you want to add this resource to destroy, this installer should either delete or fail... i'm against the half-way skip on permission.

Reply:
OK. I will change it to either delete or fail and put the onus on the Hive user to ensure that the credentials that they provide has the necessary permissions to delete the VPC Peering Connection.